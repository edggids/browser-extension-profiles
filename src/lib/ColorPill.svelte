<script lang="ts">
import {createEventDispatcher} from "svelte";
import type {ColorModel} from "../models/ColorModel";

export let color:ColorModel;
export let label:string|null = null;

let className = '';
export { className as class };

const dispatch = createEventDispatcher();

$:wrapperStyle = `--color-current-light: ${color.light}`;
$:wrapperClass = `inline-flex items-center p-2 hover:bg-current-light ${className}`;

function onClick() { dispatch('click'); }
</script>

<span role="button" on:click={onClick} class={wrapperClass} style={wrapperStyle}>
    <span class="inline-flex mr-3">
        <span style="width: 16px; height:24px; background-color: {color.light}"></span>
        <span style="width: 16px; height:24px; background-color: {color.default}"></span>
        <span style="width: 16px; height:24px; background-color: {color.darker}"></span>
    </span>

    {#if label}
        <span class="text leading-none pr-3">{label}</span>
    {/if}
</span>
