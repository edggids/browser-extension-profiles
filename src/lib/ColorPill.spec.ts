import {cleanup, render} from '@testing-library/svelte';
import {describe, it, expect, afterEach, vi} from 'vitest';
import ColorPill from "./ColorPill.svelte";

describe('ColorPill component', () => {
  afterEach(() => {
    cleanup();
  });

  it('mounts', () => {
    const { container } = render(ColorPill, { props: { color: { light: '#fff', default: '#000', darker: '#333' } } });

    expect(container).toMatchSnapshot();
  });

  it('renders with custom label', () => {
    const { container } = render(ColorPill, { props: { color: { light: '#fff', default: '#000', darker: '#333' }, label: 'Test Label' } });
    const pill = container.querySelector('span');

    expect(pill.textContent).toContain('Test Label');
    expect(pill).toMatchSnapshot();
  });

  it('dispatches click event', async () => {
    const { container, component } = render(ColorPill, { props: { color: { light: '#fff', default: '#000', darker: '#333' } } });
    const pill = container.querySelector('span');

    const onClick = vi.fn();
    component.$on('click', onClick)

    await pill.click();
    expect(onClick).toHaveBeenCalled();
  })

  it('accepts custom class', () => {
    const { container } = render(ColorPill, { props: { color: { light: '#fff', default: '#000', darker: '#333' }, class: 'custom-class' } });
    const pill = container.querySelector('span');

    expect(pill.classList.contains('custom-class')).toBe(true);
  })
});
