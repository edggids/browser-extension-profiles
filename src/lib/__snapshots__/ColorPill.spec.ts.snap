// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ColorPill component > dispatches click event 1`] = `
<span
  class="inline-flex items-center p-2 hover:bg-current-light "
  role="button"
  style="--color-current-light: #fff"
>
  <span
    class="inline-flex mr-3"
  >
    <span
      style="width: 16px; height: 24px; background-color: rgb(255, 255, 255);"
    />
     
    <span
      style="width: 16px; height: 24px; background-color: rgb(0, 0, 0);"
    />
     
    <span
      style="width: 16px; height: 24px; background-color: rgb(51, 51, 51);"
    />
  </span>
   
</span>
`;

exports[`ColorPill component > mounts 1`] = `
<body>
  <div>
    <span
      class="inline-flex items-center p-2 hover:bg-current-light "
      role="button"
      style="--color-current-light: #fff"
    >
      <span
        class="inline-flex mr-3"
      >
        <span
          style="width: 16px; height: 24px; background-color: rgb(255, 255, 255);"
        />
         
        <span
          style="width: 16px; height: 24px; background-color: rgb(0, 0, 0);"
        />
         
        <span
          style="width: 16px; height: 24px; background-color: rgb(51, 51, 51);"
        />
      </span>
       
    </span>
  </div>
</body>
`;

exports[`ColorPill component > renders with custom label 1`] = `
<span
  class="inline-flex items-center p-2 hover:bg-current-light "
  role="button"
  style="--color-current-light: #fff"
>
  <span
    class="inline-flex mr-3"
  >
    <span
      style="width: 16px; height: 24px; background-color: rgb(255, 255, 255);"
    />
     
    <span
      style="width: 16px; height: 24px; background-color: rgb(0, 0, 0);"
    />
     
    <span
      style="width: 16px; height: 24px; background-color: rgb(51, 51, 51);"
    />
  </span>
   
  <span
    class="text leading-none pr-3"
  >
    Test Label
  </span>
</span>
`;
